html_css/v2/main-course/selectors.md

```diff
@@ -0,0 +1,264 @@
+
+[`::first-letter`](https://css-tricks.com/almanac/selectors/f/first-letter/) and [`::first-line`](https://css-tricks.com/almanac/selectors/f/first-line/) allow you to (you guessed it!) give special styling to the first letter or line of some text.
+
+[`::before` and `::after`](https://css-tricks.com/almanac/selectors/a/after-and-before/) allow us to add extra elements onto the page with CSS, instead of HTML. Using it to decorate text in various ways is one common use case.
+
+~~~html
+<style>
+  .emojify::before {
+    content: '😎 🥸 🤓';
+}
+
+  .emojify::after {
+    content: '🤓 🥸 😎';
+}
+</style>
+
+<body>
+  <div> Let's <span class="emojify">emojify</span>this span!</div>
+</body>
+~~~
+
+Using these pseudo-elements this way would give us this result:
+
+Let's 😎 🥸 🤓 emojify 🤓 🥸 😎 this span!
+
+[`::selection`](https://css-tricks.com/almanac/selectors/s/selection/) allows you to change the highlighting when a user selects text on the page.
+
+ There are lots more! Have a quick browse through the [pseudo-element docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements) to see a fuller list of what's possible.
+
+### Attribute Selectors
+The last tool we're going to add to the box is attribute selectors. Recall that an attribute is simply anything in the opening tag of an HTML element - such as `img='picture.jpg'` or `href="www.theodinproject.com`.
+
+Since we write our own values for attributes, we need a slightly more flexible system to be able to target specific values.
+
+Attribute selectors have the same specificity as classes and pseudo-classes (0,0,1,0).
+
+Let's look at some examples for basic usage.
+
+* `[attribute]` - This general selector will select anything where the given attribute exists. Its value doesn't matter.
+* `selector[attribute]` - Optionally we can combine our attribute selectors with other types of selectors, such as class or element selectors.
+* `[attribute="value"]` - To get really specific, we can use `=` to match a specific attribute with a specific value.
+
+~~~css
+  [src] {
+    /* This will target any element that has a src attribute. */
+  }
+
+  img[src] {
+    /* This will only target img elements that have a src attribute. */
+  }
+
+  img[src="puppy.jpg"] {
+    /* This will target img elements with a src attribute that is exactly "puppy.jpg" */
+  }
+~~~
+
+Sometimes we need to be more general in how we access these attributes. For example, perhaps we're only interested in `img` elements where the `src` attribute's value ends in `.jpg`. For cases like this we have some attribute selectors that allow us to match a part of the attribute's value. If you've ever come across [regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) before, these attributes use a similar syntax.
+
+
+* `[attribute^="value"]` - `^=` Will match strings from the start.
+* `[attribute$="value"]` - `$=` Will strings from the end.
```

Comment:
I think line 207 should say "Will match strings from the start"

Reply:
Good catch, thanks!